HelpMyStreetFE/HelpMyStreetFE/src/js/coveragemap.js

```diff
@@ -368,29 +377,41 @@ function getDistanceInMeters(lat1, lon1, lat2, lon2) {
 
 function addMarker(marker) {
     let key = getMarkerKey(marker);
-
-    if (!googleMapMarkers.has(key)) {
-
-        marker.addListener('click', function () {
-            setMapCentre(marker.getPosition().lat(), marker.getPosition().lng(), googleMap.getZoom() + 1);
-        });
-
+    let alreadyExists = googleMapMarkers.has(key)
+    if (!alreadyExists) {
+        console.log(marker.title);
         googleMapMarkers.set(key, marker);
     }
 }
 
 function getMarkerKey(marker) {
-    return marker.getPosition().lat() + '_' + marker.getPosition().lng();
+    if (marker.type == "community"){
+        return `community_${marker.title}`;
+    } else {
+        return marker.getPosition().lat() + '_' + marker.getPosition().lng();
+    }
 }
 
 function setMapOnAll(googleMap) {
     googleMapMarkers.forEach(function (value, key, mapCollection) {
+        var onMap = value.getMap() != undefined;
+        console.log("onMap: ", onMap);
+        if (!onMap){
         value.setMap(googleMap);
+        }
     });
 }
 
 function clearMarkers() {
-    setMapOnAll(null);
+  googleMapMarkers.forEach(function (value, key, mapCollection) {
+    if (value.type == "community") {
+      //value.setMap(null);
```

Comment:
Commented out code could be confusing.  Suggest either leaving a comment along the lines of
`// Do nothing because...`
Or inverting the if statement